### ISSUE TITLE
Save/Update-Help -Module parameter should have argument completer

### ISSUE THREAD

**Issue by ThomasNieto** [PowerShell/PowerShell#20431]
### Summary of the new feature / enhancement

Save/Update-Help Module parameter should have an argument completer.

### Proposed technical implementation details (optional)

Can use the existing module name completer from `Get-Command -Module`.

**Comment by jhoneill** [comment]
I'm seeing quite a lot of these argument completer issues being opened. IMHO everything that can possibly have a completer should have so I'm all for it. BUT would it make sense to have a piece of work within the team to check that all the commands which ship _as part of the language_ have completers instead of opening each missing one as its own issue ? This is (a) a way of improving the quality and consistency of the experience and (b) setting a good example to developers.

**Comment by kilasuit** [comment]
Commenting as myself and not on behalf of the `Interactive-UX` Working Group 

This is something that I'd like to see for productivity from an interactive perspective & I have 2 main thoughts on this other than the proposed implementation makes sense to me 

1) we can either add this natively so it's always there (irrespective of whether interactive or not) 
2) or does it make more sense to externalise all argument completers into a module like `TabExpansion++` and then ship it like we do other PowerShell modules from the Gallery and then when we open an Interactive session we load that module like we do PSReadline and don't when `pwsh -noprofile -noninteractive` is called.

If we externalise this then in future we can reduce the overall package size of PowerShell (which is still complained about) with a [lite package](https://github.com/PowerShell/PowerShell/issues/5681) which I know has been something that has been discussed many a time before

I know that's beyond the ask here, but if we externalised this then we could also backport too which would be an additional hidden benefit and like @jhoneill mentions may be an easier way to manage this as a bigger improvement piece but at minimum could be done either as a mega issue or project in here

**Comment by ThomasNieto** [comment]
@kilasuit my opinion is core PowerShell cmdlets should have argument completers in-box and not a separate module. See #17283   that importing argument completers via a module causes PowerShell startup time to suffer. The argument completer issues I've opened already exist in `TabExpansion++` but could no longer handle the ~10+ second penalty.

The size and performance penalty to having these core argument completers defined in C# is minimal.

**Comment by sdwheeler** [comment]
The following cmdlets have `-Module` parameters:

| Cmdlet | Has completer |
| --- | --- |
| Get-Command | Yes |
| Export-PSSession | No |
| Import-PSSession | No |
| Save-Help | No |
| Update-Help | No |

The following module cmdlets have a `-Name` parameter:

| Cmdlet | Has completer |
| --- | --- |
| Get-Module | Yes |
| Import-Module | Yes |
| New-Module | No - but shouldn't |
| Remove-Module | Yes |

The question is should completers be added to cmdlets that don't have argument completers to resolve module names?

For consistency, I think `Save-Help` and `Update-Help` should have completers.

I think the PSSession cmdlet should not.

**Comment by kilasuit** [comment]
@ThomasNieto - I don't necessarily disagree & personally prefer to have them defined in the cmdlet/function definition where possible unless there is an element of potential reusability of the completer like in this case.  elsewhere is something that's desired and shipped inbox & where possible suggest when I've trained others that this is best done as part of the the first phase of cmdlet/function development and included when v1 of a module is released, but that's not always a possibility due to time constraints.

Just pointing out additional benefit that could be gained by taking a different course than taking the quick win option. 

There would be nothing stopping a possible rewrite in c# of the completers that are available in TE++ similar to [this comment from @lzybkr](https://github.com/lzybkr/TabExpansionPlusPlus/issues/78#issuecomment-523727242) as a new module that could potentially be optimised in how it's loaded when running in an interactive session only but would also require work in engine to make that happen. 

However that's another PowerShellGet esk size task which is why I'd rather the WG's discuss both short and longer term approaches here. Interactive-UX should be discussing this today btw!

So to sum up my thoughts on which ways we could go with this more for referencing later on as part of any other argument completer conversations going forward

Quick Wins/Short term/least effort - we add this argument completer (& any others found desirable) directly to this repo for any and all core cmdlets 
Longer term/most effort - build a new c# module with argument completers added from te++ ship that inbox in future and potentially also allow to be made available for down level support too.

**Comment by theJasonHelmick** [comment]
Thank you @ThomasNieto  for reporting this issue.  The Interactive WG has reviewed this issue and agrees that this should be fixed in future releases.  It would benefit the team to open this up to the community as "Up-For-Grabs".
